June releases announcement blog (#193)
conda-incubator/conda-dot-org: blog/2024-06-14-june-releases.mdx

---
title: "June 2024 Releases"
slug: "2024-06-14-june-releases"
authors: [dashagurova]
tags:
  [announcement, conda-index, conda-package-handling, conda-package-streaming]
description: |
  Conda-index 0.5.0, conda-package-handling 2.3.0 and conda-package-streaming 0.10.0 have been released! 🎉
image: img/blog/2024-06-14-june-releases/banner.png
---

> The June 2024 releases included updates to conda-index, conda-package-streaming and conda-package-handling! 🎉 All of these new releases have been released to both `main` and `conda-forge`.

Reviewer: I was thinking it might be good to include links to the repos in case people aren't as familiar with these packages

```suggestion
The June 2024 releases included updates to [conda-index](https://github.com/conda/conda-index), [conda-package-streaming](https://github.com/conda/conda-package-streaming) and [conda-package-handling](https://github.com/conda/conda-package-handling)! 🎉 All of these new releases have been released to both `main` and `conda-forge`.
```

## Changes in conda-index [0.5.0](https://github.com/conda/conda-index/releases/tag/0.5.0)

To update conda-index to the latest version, run:

```bash
conda install -n base conda-index=0.5.0
```

<!-- truncate -->

### ✨ What's New? ✨

> - Add experimental python `-m conda_index.json2jlap` script to run after indexing, to create `repodata.jlap` patch sets for incremental repodata downloads.

Reviewer 2: ```suggestion
- Add experimental `python -m conda_index.json2jlap` script to run after indexing, to create `repodata.jlap` patch sets for incremental repodata downloads.
```

- Add `--current-repodata/--no-current-repodata` flags to control whether
  `current_repodata.json` is generated.

> - Add support for CEP-15 `base_url` to host packages separate from repodata.

Reviewer: ```suggestion
- Add support for [CEP-15](https://github.com/conda/ceps/blob/main/cep-15.md) `base_url` to host packages separate from repodata.
```

> - Support fsspec (in the API only) to index any fsspec-supported remote

Reviewer: ```suggestion
- Support `fsspec` (in the API only) to index any `fsspec`-supported remote
```

  filesystem. Also enables the input packages folder to be separate from the
  cache and output folders.

### 🔧 What Got Fixed? 🔧

- Move `run_exports.json` query into cache, instead of directly using SQL in
  `ChannelIndex`.
- Create parents when creating `<subdir>/.cache`.

### ⚙️ Other Updates ⚙️

> - Approach 100% code coverage in test suite; reformat with ruff.

Reviewer: ```suggestion
- Approach 100% code coverage in test suite; reformat with `ruff`.
```

- Update CI configuration to test on more platforms.
- Drop support for Python 3.7; support Python 3.8+ only.

## Changes in conda-package-handling [2.3.0](https://github.com/conda/conda-package-handling/releases/tag/2.3.0)

To update conda-package-handling to the latest version, run:

```bash
conda install -n base conda-package-handling=2.3.0
```

<!-- truncate -->

### ✨ What's New? ✨

- Add `cph list` to report artifact contents without prior extraction.
- Added formal support for Python 3.10, 3.11, and 3.12.

### 🔧 What Got Fixed? 🔧

- Delay `os.getcwd()` call to body of `CondaFormat_v2.create()` when `out_folder` is not passed.

### 🌅 Deprecations 🌅

- Removed formal support for Python 3.7.

### ⚙️ Other Updates ⚙️

> - Remove MANIFEST.in, used for Python sdists, which referenced non-existent files. Source distributions appear correct without MANIFEST.in.

Reviewer: ```suggestion
- Remove `MANIFEST.in`, used for Python `sdist`s, which referenced non-existent files. Source distributions appear correct without `MANIFEST.in`.
```

- Add explicit `zstandard` dependency.

## Changes in conda-package-streaming [0.10.0](https://github.com/conda/conda-package-streaming/releases/tag/v0.10.0)

To update conda-package-streaming to the latest version, run:

```bash
conda install -n base conda-package-streaming=0.10.0
```

<!-- truncate -->

### ✨ What's New? ✨

> - Use zip64 extensions when converting .tar.bz2 to .conda, if uncompressed size is close to the 2GB ZIP64_LIMIT.

Reviewer: ```suggestion
- Use `zip64` extensions when converting `.tar.bz2` to `.conda`, if uncompressed size is close to the 2GB `ZIP64_LIMIT`.
```
